content/_code-samples/tx-serialization/js/main.js

```diff
@@ -0,0 +1,45 @@
+const fs = require('fs');
+const { Client } = require("xrpl");
+const ripple_binary_codec = require('ripple-binary-codec');
+
+const payload = require("./json_input.json");
+let outputs = require("./binary_outputs.json");
+
+// NOTE: This sample uses ripple-binary-codec (https://github.com/XRPLF/xrpl.js/tree/main/packages/ripple-binary-codec) and xrpl.js (https://github.com/XRPLF/xrpl.js/) libraries
+
+const main = async () => {
+    try {
+        console.log("Converting your transaction...");
+
+        // connect to a public server
+        const client = new Client("wss://xrplcluster.com/");
+        await client.connect();
+
+        // autofill all the missing fields
+        const complete_payload = await client.autofill(payload);
+
+        // convert the JSON tx to serialized using ripple-binary-codec
+        const decoded_payload = ripple_binary_codec.encode(complete_payload);
+
+        // push binary tx into output array
+        outputs.push(decoded_payload);
+
+        // write the output into binary_outputs.json file.
+        fs.writeFileSync("./binary_outputs.json", JSON.stringify(outputs, null, "\t"));
```

Comment:
Why write to file instead of just printing the result via `console.log()`?

Reply:
Good point, I will change it right away.